play.pokemonshowdown.com/src/battle-log.ts

```diff
@@ -353,20 +353,44 @@ export class BattleLog {
 			break;
 
 		case 'turn':
+			const goToTurnButton = document.createElement('button');
 			const h2elem = document.createElement('h2');
+			goToTurnButton.className = 'subtle';
 			h2elem.className = 'battle-history';
 			let turnMessage;
+			let turnNumber: number;
 			if (this.battleParser) {
 				turnMessage = this.battleParser.parseArgs(args, {}).trim();
 				if (!turnMessage.startsWith('==') || !turnMessage.endsWith('==')) {
 					throw new Error("Turn message must be a heading.");
 				}
 				turnMessage = turnMessage.slice(2, -2).trim();
+				turnNumber = parseInt(turnMessage.split(' ')[1].trim());
 				this.battleParser.curLineSection = 'break';
 			} else {
 				turnMessage = `Turn ${args[1]}`;
+				turnNumber = parseInt(args[1]);
 			}
-			h2elem.innerHTML = BattleLog.escapeHTML(turnMessage);
+			let userId: string | undefined = undefined;
+			if (window.app?.user?.get) {
+				userId = window.app.user.get('userid');
+			}
+			if (!this.scene?.battle.isReplay && userId && [
+				this.scene?.battle.p1.id,
+				this.scene?.battle.p2.id,
+				this.scene?.battle.p3?.id,
+				this.scene?.battle.p4?.id,
+			].includes(userId)) {
```

Comment:
```suggestion
			if (this.scene?.battle.myPokemon) {
```

I'm 80% sure this works. Can you test it and tell me if it does?

Reply:
So I tested this originally but I found that there is strange behavior with when myPokemon is null. For example, if you refresh the browser mid battle it will temporarily be null which will then add the button to the headers. So theoretically, if you played straight through the battle and never refreshed or clicked off it would be fine but if you did you would wind up with mixed headers (some being buttons, some not).
Also when you did refresh, any headers that were not buttons will become buttons up to the current turn.

I also wrote this solution off because it felt like an "indirect" solution, if that makes sense?

Reply:
Yeah that makes sense. Frustrating. On the other hand we do want battlers to be able to go to turn after the game is over. Probably the solution is that they should always be buttons, but they should be unclickable while the battle is ongoing.

Reply:
I think that is probably the best way forward. 

My knowledge of the code in its entirety is a little limited so I might be wrong here, but from my understanding: the battlelog has elements appended to it (like messages or turn headers) which are rendered on the DOM.
If this is the case then the turn headers would not be updated after the battle is finished unless the instant replay button was clicked?
Unless we wanted to, upon the battle ending, call something that goes and updates the turn headers I'm not sure how the button would become clickable without re-rendering the log by clicking the instant replay button.

Reply:
The simplest would be to make it always clickable and simply give an error to players in a battle.

Reply:
A more complex way would be go and update the turn headers with a script.

Reply:
I guess my thought process here is: why make it clickable to give an error for the users when it is already set to be non-clickable for users in battle? This adds extra clutter when the current change accomplishes the same thing without the extra noise. To be able to click the turn button, the players will have to click the replay button either way (excluding a script that updates the turn header)